presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriter.java

```diff
@@ -49,13 +52,16 @@
         implements HiveFileWriter
 {
     private static final int INSTANCE_SIZE = ClassLayout.parseClass(OrcFileWriter.class).instanceSize();
+    private static final ThreadMXBean THREAD_MX_BEAN = ManagementFactory.getThreadMXBean();
 
     private final OrcWriter orcWriter;
     private final Callable<Void> rollbackAction;
     private final int[] fileInputColumnIndexes;
     private final List<Block> nullBlocks;
     private final Optional<Supplier<OrcDataSource>> validationInputFactory;
 
+    private final AtomicLong validationCpuNanos = new AtomicLong();
```

Comment:
I'm pretty sure all of this code is single threaded, but I'm not sure about this callers of this method.  If this method is allowed to be called concurrently, it should be documented.

Reply:
@dain : I believe `Operator.finish` will only be called in single-threaded. Thus this method cannot be called concurrently.